test/Microsoft.ML.Functional.Tests/ONNX.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Microsoft.ML.Functional.Tests.Datasets;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.TestFramework.Attributes;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Trainers.FastTree;
+using Microsoft.ML.Transforms;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Functional.Tests
+{
+    public class ONNX : BaseTestClass
+    {
+        public ONNX(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <summary>
+        /// ONNX: I can save a model to ONNX and reload it and use it in a pipeline.
+        /// </summary>
+        [OnnxFactAttribute]
+        public void SaveOnnxModelLoadAndScoreFastTree()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            // Get the dataset.
+            var data = mlContext.Data.LoadFromTextFile<HousingRegression>(GetDataPath(TestDatasets.housing.trainFilename), hasHeader: true);
+
+            // Create a pipeline to train on the housing data.
+            var pipeline = mlContext.Transforms.Concatenate("Features", HousingRegression.Features)
+                .Append(mlContext.Transforms.Normalize("Features"))
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.Regression.Trainers.FastTree(
+                    new FastTreeRegressionTrainer.Options { NumberOfThreads = 1, NumberOfTrees = 10 }));
+
+            // Fit the pipeline.
+            var model = pipeline.Fit(data);
+
+            // Serialize the pipeline to a file.
+            var modelFileName = "SaveOnnxLoadAndScoreFastTreeModel.onnx";
+            var modelPath = DeleteOutputPath(modelFileName);
+            using (var file = File.Create(modelPath))
+                mlContext.Model.ConvertToOnnx(model, data, file);
+
+            // Load the model as a transform.
+            var onnxEstimator = mlContext.Transforms.ApplyOnnxModel(modelPath);
+            var onnxModel = onnxEstimator.Fit(data);
+
+            // TODO #2980: ONNX outputs don't match the outputs of the model, so we must hand-correct this for now.
+            // TODO #2981: ONNX models cannot be fit as part of a pipeline, so we must use a workaround like this.
+            var onnxWorkaroundPipeline = onnxModel.Append(
+                mlContext.Transforms.CopyColumns("Score", "Score0").Fit(onnxModel.Transform(data)));
```

Comment:
>Score0 [](start = 59, length = 6)

Just a question: where does `Score0` come from? is it produced by onnx transform?

Reply:
I am not sure where the '0' suffices are entering the pipeline yet. I am digging into it as part of this bug: #2980.

---
In reply to: [267067407](https://github.com/dotnet/machinelearning/pull/2984#discussion_r267067407) [](ancestors = 267067407)